### PR TITLE
test add octave,scilab backend

### PR DESCRIPTION
**Pull request by kevinsmia1939**
test add octave backend

### CODE REVIEW

**Comment by flyingcakes85** [comment]
Looks like we're missing `.gitmodules`. 

See this file -> https://github.com/flyingcakes85/flathub/blob/org.kde.kwave/.gitmodules

**Comment by kevinsmia1939** [comment]
bot, build

File: org.kde.cantor.json [hunk 1]
```diff
@@ -13,12 +13,25 @@
         "--socket=fallback-x11",
         "--socket=wayland",
         "--filesystem=xdg-documents",
-        "--device=dri"
+        "--device=dri",
+        "--socket=pulseaudio",
+        "--share=network",
+        "--filesystem=host",
```
Comment (travier) [inline org.kde.cantor.json:19]: Is this one really needed? Should it be home instead?

File: python3-requirements.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                },
+                {
+                    "type": "file",
+                    "url": "https://files.pythonhosted.org/packages/46/3e/fbe19f658d396d52975413cd2718fbada15de5d3fa2892e23e1aca6ffed4/sympy-1.5.1.tar.gz",
+                    "sha256": "d77901d748287d15281f5ffe5b0fef62dd38f357c2b827c44ff07f35695f4e7e"
+                }
+            ]
+        }
+    ]
+}
```
Comment (travier) [inline python3-requirements.json:40]: Minor: Missing new line here

**Review by travier — COMMENTED** [review]
Most of it comes from the Octave Flatpak and the permissions are lightly explained in https://github.com/flathub/org.octave.Octave/blob/master/org.octave.Octave.yaml#L17.

**Comment by travier** [comment]
I'm wondering if it would make sense to suggest to the folks packaging the Octave Flatpak to make it so that we could directly include it as a submodule.

**Comment by kevinsmia1939** [comment]
> I'm wondering if it would make sense to suggest to the folks packaging the Octave Flatpak to make it so that we could directly include it as a submodule.

Do you mean to include it to shared-modules?

**Comment by kevinsmia1939** [comment]
I test it, it found Octave now.

**Comment by kevinsmia1939** [comment]
test adding scilab

**Comment by kevinsmia1939** [comment]
I can't figure out how to build Scilab from source, so I use the prebuild, but it would not have arm support.

**Comment by travier** [comment]
> > I'm wondering if it would make sense to suggest to the folks packaging the Octave Flatpak to make it so that we could directly include it as a submodule.
> 
> Do you mean to include it to shared-modules?

No, I mean that we would include their repo as a submodule and directly include their manifest.

**Comment by kevinsmia1939** [comment]
> > > I'm wondering if it would make sense to suggest to the folks packaging the Octave Flatpak to make it so that we could directly include it as a submodule.
> > 
> > 
> > Do you mean to include it to shared-modules?
> 
> No, I mean that we would include their repo as a submodule and directly include their manifest.

I tried to use the file org.octave.Octave.yaml at first, but it is not in correct format because it have app-id and other stuff so it does not work.

File: python3-requirements.json (hunk 2 above)
Comment (kevinsmia1939) [reply python3-requirements.json:40]: I don't really understand.

**Comment by kevinsmia1939** [comment]
would it be ok to leave sage out for now, dont know how to build it.

**Comment by travier** [comment]
> I tried to use the file org.octave.Octave.yaml at first, but it is not in correct format because it have app-id and other stuff so it does not work.

Yes, the idea would be that they split their manifest in two: the modules in an independent one and the sdk, etc. info in the main one. This way we would be able to include their other manifest directly.

File: python3-requirements.json (hunk 2 above)
Comment (travier) [reply python3-requirements.json:40]: There is no empty line at the end of the file. This is mostly a convention and not a bug.

**Comment by kevinsmia1939** [comment]
> > I tried to use the file org.octave.Octave.yaml at first, but it is not in correct format because it have app-id and other stuff so it does not work.
> 
> Yes, the idea would be that they split their manifest in two: the modules in an independent one and the sdk, etc. info in the main one. This way we would be able to include their other manifest directly.

How about using some script to remove the problematic part and convert it to submodule?

**Comment by travier** [comment]
Sure, works for me too.

**Comment by travier** [comment]
Thinking about this more, this won't work. When would the script run?

**Comment by travier** [comment]
https://github.com/flathub/org.octave.Octave/issues/209

**Pull request closed without merging**